react/features/overlay/reducer.js

```diff
@@ -0,0 +1,140 @@
+
+    case SUSPEND_DETECTED:
+        return _suspendDetected(state, action);
+    }
+
+    return state;
+});
+
+/**
+ * Reduces a specific Redux action CONFERENCE_FAILED of the feature overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @param {Action} action - The Redux action CONFERENCE_FAILED to reduce.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _conferenceFailed(state, action) {
+    const ConferenceErrors = JitsiMeetJS.errors.conference;
+
+    if (action.error === ConferenceErrors.FOCUS_LEFT
+            || action.error === ConferenceErrors.VIDEOBRIDGE_NOT_AVAILABLE) {
+        return setStateProperties(state, {
+            haveToReload: true,
+            isNetworkFailure: false,
+            reason: action.errorMessage
+        });
+    }
+
+    return state;
+}
+
+/**
+ * Reduces a specific Redux action CONNECTION_ESTABLISHED of the feature
+ * overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _connectionEstablished(state) {
+    return setStateProperty(state, 'connectionEstablished', true);
+}
+
+/**
+ * Reduces a specific Redux action CONNECTION_FAILED of the feature overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @param {Action} action - The Redux action CONNECTION_FAILED to reduce.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _connectionFailed(state, action) {
+    const ConnectionErrors = JitsiMeetJS.errors.connection;
+
+    switch (action.error) {
+    case ConnectionErrors.CONNECTION_DROPPED_ERROR:
+    case ConnectionErrors.OTHER_ERROR:
+    case ConnectionErrors.SERVER_ERROR:
+        logger.error(`XMPP connection error: ${action.errorMessage}`);
+
+        // From all of the cases above only CONNECTION_DROPPED_ERROR
+        // is considered a network type of failure
+        return setStateProperties(state, {
+            haveToReload: true,
+            isNetworkFailure:
+                action.error === ConnectionErrors.CONNECTION_DROPPED_ERROR,
+            reason: `xmpp-conn-dropped: ${action.errorMessage}`
+        });
+    }
+
+    return state;
+}
+
+/**
+ * Reduces a specific Redux action MEDIA_PERMISSION_PROMPT_VISIBILITY_CHANGED of
+ * the feature overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @param {Action} action - The Redux action to reduce.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _mediaPermissionPromptVisibilityChanged(state, action) {
+    return setStateProperties(state, {
+        mediaPermissionPromptVisible: action.isVisible,
```

Comment:
I'm afraid I don't follow the `boolean` variable naming style followed in this PR. Regardless of the side on which I stand, do you or do you not want to have `is`? I realize that `is` encodes the `boolean` type into the variable name but you're not using it always in your PR so I'm not sure whether you're taking advantage of the name, whether the rule is more complex than simply `typeof isXXX === 'boolean'`, or there's merely no rule.